woocommerce/woocommerce-ios
# Dynamic Dashboard: Coupons Card UI

Part of: #12716


## Description


Adds a first iteration of UI related code for the "Most Active Coupons" dashboard card. 

Changes
- View and View model for  "Most Active Coupons" dashboard card.
- Way to store and retrieve selected time range to App settings.
- Open the coupons list screen.
- Update coupon summary text format to match new design.

## Testing instructions


- Enable the feature flag `dynamicDashboardM2` and build the app.
- Confirm that the dashboard still has the same layout as before.
- Tap Edit button and confirm that there's an option for the most active coupons card and it's disabled.
- Enable coupons card and save the change.
- Confirm that the coupons card is displayed on the dashboard.
- Confirm that the UI matches the Figma design.
- Tap View all coupons, confirm that coupons list screen is displayed.
- Tap the calendar icon and change time range, validate that time range selection works.
- Tap the ellipsis menu on the coupons card, confirm that there's an option Hide Coupons. 
- Tap Hide coupons, confirm that the customize screen is displayed.

Disable the `dynamicDashboardM2` flag and confirm that the Coupons card is not available.

## Screenshots


| Light | Dark |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-14 at 17 10 13](https://github.com/woocommerce/woocommerce-ios/assets/524475/c47d9e00-cfbd-445a-954e-3ab406b227b9) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-14 at 17 10 39](https://github.com/woocommerce/woocommerce-ios/assets/524475/c6632d38-88d1-4e27-9102-acf27f59c608) | 

Time range selection and loading 
![Simulator Screen Recording - iPhone 15 Pro Max - 2024-05-14 at 17 11 57](https://github.com/woocommerce/woocommerce-ios/assets/524475/6559a178-2086-4988-a6c1-42cd982522b6)

---
